test/composer.test.ts

```diff
@@ -1,20 +1,51 @@
-import { Composer, type MiddlewareFn } from "../src/composer.ts";
-import { type Context } from "../src/mod.ts";
+import { BotError, Composer, type MiddlewareFn, run } from "../src/composer.ts";
+import { Context } from "../src/mod.ts";
 import {
     assertEquals,
+    assertFalse,
+    assertInstanceOf,
     assertRejects,
+    assertStringIncludes,
     beforeEach,
     describe,
     it,
     type Spy,
     spy,
 } from "./deps.test.ts";
 
+describe("BotError", () => {
+    // @ts-ignore-error this message is missing many properties
+    const ctx = new Context({ message: { text: "test" } }, 0, 0);
+    it("should copy stack and message", () => {
+        const e = new Error("nope");
+        const err = new BotError(e, ctx);
+        assertStringIncludes(err.message, e.message);
+        assertEquals(err.stack, e.stack);
+    });
+    it("should find good error message for non-error objects", () => {
+        const e = (value: unknown) => new BotError(value, ctx).message;
+        assertStringIncludes(e("test"), "test");
+        assertStringIncludes(e("test"), "of type string");
+        assertFalse(e(Array(50).fill("a").join("") + "bbb").includes("b")); // cut long strings
+        assertStringIncludes(e(42), "42");
+        assertStringIncludes(e(42), "of type number");
+        assertStringIncludes(e(() => {}), "of type function");
+    });
+});
+
 describe("Composer", () => {
     let composer: Composer<Context>;
-    const ctx = { message: { text: "" } } as Context;
+    const ctx = new Context(
+        // deno-lint-ignore no-explicit-any
```

Comment:
Again, use ts-ignore-error over `as any` and a lint comment. (I will omit the other hints to this in this file)